stdlib/public/core/ManagedBuffer.swift

```diff
@@ -34,22 +34,14 @@ open class ManagedBuffer<Header, Element> {
       ManagedBuffer<Header, Element>) throws -> Header
   ) rethrows -> ManagedBuffer<Header, Element> {
 
-    let p = try ManagedBufferPointer<Header, Element>(
-      bufferClass: self,
-      minimumCapacity: minimumCapacity,
-      makingHeaderWith: { buffer, _ in
-        try factory(
-          unsafeDowncast(buffer, to: ManagedBuffer<Header, Element>.self))
-      })
-
-    return unsafeDowncast(p.buffer, to: ManagedBuffer<Header, Element>.self)
-  }
+    let p = Builtin.allocWithTailElems_1(
+         self,
+         minimumCapacity._builtinWordValue, Element.self)
 
-  /// Destroy the stored Header.
-  deinit {
-    ManagedBufferPointer(self).withUnsafeMutablePointerToHeader {
-      _ = $0.deinitialize()
-    }
+    let initHeaderVal = try factory(p)
+    p.headerAddress.initialize(to: initHeaderVal)
+    _fixLifetime(p) 
```

Comment:
I just talked to Andy. Actually we would not need the _fixLIfetime here because even if p is not used after inlining, the optimizer would not shrink the lifetime of p.
But, as Dave wrote, we should not try to optimize the fixLifetime away.